packages/flutter/lib/src/widgets/view.dart

```diff
@@ -106,13 +146,593 @@ class View extends StatelessWidget {
+    assert(slot == null);
+    assert(renderObject.debugValidateChild(child));
+    renderObject.child = child;
+  }
+
+  @override
+  void moveRenderObjectChild(RenderObject child, Object? oldSlot, Object? newSlot) {
+    assert(false);
+  }
+
+  @override
+  void removeRenderObjectChild(RenderObject child, Object? slot) {
+    assert(slot == null);
+    assert(renderObject.child == child);
+    renderObject.child = null;
+  }
+
+  @override
+  void unmount() {
+    if (_effectivePipelineOwner != (widget as RawView)._deprecatedPipelineOwner) {
+      _effectivePipelineOwner.dispose();
+    }
+    super.unmount();
+  }
 }
 
 class _ViewScope extends InheritedWidget {
   const _ViewScope({required this.view, required super.child});
 
-  final FlutterView view;
+  final FlutterView? view;
 
   @override
   bool updateShouldNotify(_ViewScope oldWidget) => view != oldWidget.view;
 }
+
+class _PipelineOwnerScope extends InheritedWidget {
+  const _PipelineOwnerScope({
+    required this.pipelineOwner,
+    required super.child,
+  });
+
+  final PipelineOwner pipelineOwner;
+
+  @override
+  bool updateShouldNotify(_PipelineOwnerScope oldWidget) => pipelineOwner != oldWidget.pipelineOwner;
+}
+
+class _MultiChildComponentWidget extends Widget {
+  const _MultiChildComponentWidget({
+    super.key,
+    List<Widget> views = const <Widget>[],
+    Widget? child,
+  }) : _views = views, _child = child;
+
+  // It is up to the subclasses to make the relevant properties public.
+  final List<Widget> _views;
+  final Widget? _child;
+
+  @override
+  Element createElement() => _MultiChildComponentElement(this);
+}
+
+/// A collection of sibling [View]s.
+///
+/// This widget can only be used in places were a [View] widget is allowed. In
+/// practical terms, it can be used at the root of the widget tree outside of
+/// any [View] widget, as a child to a another [ViewCollection], or in the
+/// [ViewAnchor.view] slot of a [ViewAnchor] widget. It is not required to be a
+/// direct child of those widgets; other non-[RenderObjectWidget]s may appear
+/// in between the two (such as an [InheritedWidget]).
+///
+/// Similarly, the [views] children of this widget must be [View]s, but they
+/// may be wrapped in additional non-[RenderObjectWidget]s (e.g.
+/// [InheritedWidget]s).
+class ViewCollection extends _MultiChildComponentWidget {
+  /// Creates a [ViewCollection] widget.
+  ///
+  /// The provided list of [views] must contain at least one widget.
+  const ViewCollection({super.key, required super.views}) : assert(views.length > 0);
+
+  /// The [View] descendants of this widget.
+  ///
+  /// The [View]s may be wrapped in other non-[RenderObjectWidget]s (e.g.
+  /// [InheritedWidget]s). However, no [RenderObjectWidget] is allowed to appear
+  /// between the [ViewCollection] and the next [View] widget.
+  List<Widget> get views => _views;
+}
+
+/// Decorates a [child] widget in a surrounding [View] with a side-[View].
```

Comment:
I find the one-line doc hard to understand. What do you mean by "a surrounding view", is it the same thing as the "side-view"? If they're different, why must the "surrounding view" emphasized in the first sentence? Why do we use the "decorate", is the child widget affected by the side-view?

If I understand correctly, what do you think of this one?
```suggestion
/// A widget that displays the [child] widget as-is, but also branches out a side-[View].
```